azurerm/resource_arm_mariadb_server.go

```diff
@@ -0,0 +1,381 @@
+			storageMB, _ := diff.GetOk("storage_profile.0.storage_mb")
+
+			if strings.ToLower(tier.(string)) == "basic" && storageMB.(int) > 1048576 {
+				return fmt.Errorf("basic pricing tier only supports upto 1,048,576 MB (1TB) of storage")
+			}
+
+			return nil
+		},
+	}
+}
+
+func resourceArmMariaDbServerCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).mariadbServersClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM MariaDB Server creation.")
+
+	name := d.Get("name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	adminLogin := d.Get("administrator_login").(string)
+	adminLoginPassword := d.Get("administrator_login_password").(string)
+	sslEnforcement := d.Get("ssl_enforcement").(string)
+	version := "10.2"
+	createMode := "Default"
+	tags := d.Get("tags").(map[string]interface{})
+
+	sku := expandAzureRmMariaDbServerSku(d)
+	storageProfile := expandAzureRmMariaDbStorageProfile(d)
+
+	properties := mariadb.ServerForCreate{
+		Location: &location,
+		Properties: &mariadb.ServerPropertiesForDefaultCreate{
+			AdministratorLogin:         utils.String(adminLogin),
+			AdministratorLoginPassword: utils.String(adminLoginPassword),
+			Version:                    mariadb.ServerVersion(version),
+			SslEnforcement:             mariadb.SslEnforcementEnum(sslEnforcement),
+			StorageProfile:             storageProfile,
+			CreateMode:                 mariadb.CreateMode(createMode),
+		},
+		Sku:  sku,
+		Tags: expandTags(tags),
+	}
+
+	future, err := client.Create(ctx, resourceGroup, name, properties)
+	if err != nil {
+		return fmt.Errorf("Error creating MariaDB Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for creation of MariaDB Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving MariaDB Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read MariaDB Server %q (Resource Group %q) ID", name, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmMariaDbServerRead(d, meta)
+}
+
+func resourceArmMariaDbServerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).mariadbServersClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["servers"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[WARN] MariaDB Server %q was not found (resource group %q)", name, resourceGroup)
```

Comment:
_minor:_ for consistency with the other log messages

```suggestion
			log.Printf("[WARN] MariaDB Server %q was not found (Resource Group %q)", name, resourceGroup)
```